pilot/model/config_test.go

```diff
@@ -263,6 +266,38 @@ func TestResolveHostname(t *testing.T) {
 	}
 }
 
+func TestResolveFQDN(t *testing.T) {
+	cases := []struct {
+		name string
+		meta model.ConfigMeta
+		fqdn string
+	}{
+		{
+			name: "hello",
+			meta: model.ConfigMeta{Namespace: "default", Domain: "cluster.local"},
+			fqdn: "hello.default.svc.cluster.local",
+		},
+		{
+			name: "hello.default",
+			meta: model.ConfigMeta{Namespace: "default", Domain: "cluster.local"},
+			fqdn: "hello.default.svc.cluster.local",
+		},
+		{
+			name: "hello.default.svc.cluster.local",
+			meta: model.ConfigMeta{Namespace: "default", Domain: "cluster.local"},
+			fqdn: "hello.default.svc.cluster.local",
+		},
```

Comment:
When you implement the if namespace, name+namespace+domain, else name+domain logic, add test cases appropriately.